app/src/main/res/values/styles.xml

```diff
@@ -403,7 +403,14 @@
         <item name="android:textColor">?primaryText</item>
     </style>
 
+    <style name="Body12TextStyle" parent="TextAppearance.MaterialComponents.Body1">
+        <item name="android:textColor">@color/primary_state_title_text_color</item>
```

Comment:
Sorry, I am gonna blocked this on one change here, which might've been missed in the comments https://github.com/mozilla-mobile/fenix/pull/19953#discussion_r650204484.

```suggestion
        <item name="android:textColor">@color/primary_state_title_text_color</item>
```

Reply:
This is happening in the other PR, and I can't do that until this is merged.

Reply:
Also, this is not supposed to be ?primaryText color, it's a different value

Reply:
I think there's some misunderstanding here. We want this to be `?primaryText` here since all the other BodyXXTextStyle uses `?primaryText`. 

Since this can `Body12TextStyle` is extended in `recentBookmarkItemSubTitleText`, we can just move  `<item name="android:textColor">@color/primary_state_title_text_color</item>` over there.